src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private final DateParser dateParser = new DateParser();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate = dateParser.date(dateFrom);
+        LocalDate stopDate = dateParser.date(dateTo);
+        int [] salaries = new int[names.length];
+        StringBuilder sb = new StringBuilder();
+        /*
+          Checking every date for every name;
+          Validating name; checking date to fit boundaries; calculating total salary for each date;
+          Filling StringBuilder with every valid entry separately;
+        */
+        for (int i = 0; i < names.length; i++) {
+            for (String datum : data) {
+                if (names[i].equals(datum.split(" ")[1])
+                        && dateParser.date(datum.split(" ")[0]).compareTo(startDate) >= 0
+                        && dateParser.date(datum.split(" ")[0]).compareTo(stopDate) <= 0) {
+                    salaries[i] += Integer.parseInt(datum.split(" ")[2])
+                            * Integer.parseInt(datum.split(" ")[3]);
```

Comment:
Introduce `datum.split(" ")` as local variable, no need to call `split` 6 times.

Reply:
Good point, thanks!